pkg/async/events/implementations/node_execution_event_writer.go

```diff
@@ -0,0 +1,41 @@
+package implementations
+
+import (
+	"context"
+
+	"github.com/flyteorg/flyteadmin/pkg/async/events/interfaces"
+	"github.com/flyteorg/flyteadmin/pkg/repositories"
+	"github.com/flyteorg/flyteadmin/pkg/repositories/transformers"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/logger"
+)
+
+type nodeExecutionEventWriter struct {
```

Comment:
Lets add a comment, as to why we are doing this

Reply:
done

Reply:
Why do we not have to wait - because we create the coalesced node-execution event